src/Sarif.Viewer.VisualStudio/CloseSarifLogService.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis.Sarif.Converters;
+using Microsoft.Sarif.Viewer.ErrorList; 
+
+namespace Microsoft.Sarif.Viewer
+{
+    public class CloseSarifLogService : SCloseSarifLogService, ICloseSarifLogService
+    {
+        /// <summary>
+        /// Closes all SARIF logs opened in the viewer.
+        /// </summary>
+        public void CloseAllSarifLogs()
+        {
+            ErrorListService.CloseAllSarifLogs();
+        }
+
+        /// <summary>
+        /// Closes the specified SARIF log in the viewer.
+        /// </summary>
+        /// <param name="paths">The complete path to the SARIF log file.</param>
+        public void CloseSarifLogs(IEnumerable<string> paths)
+        {
+            ErrorListService.CloseSarifLogs(paths);
+        }
+
+        public void LoadSarifLog(string path)
+        {
+            if (!string.IsNullOrWhiteSpace(path))
+            {
+                try
+                {
+                    ErrorListService.ProcessLogFile(path, SarifViewerPackage.Dte.Solution, ToolFormat.None);
+                }
+                catch (InvalidCastException) { }
```

Comment:
>InvalidCastException [](start = 23, length = 20)

Do you know under what circumstances this exception is raised? How did you know to add this handler? Code comment could be useful.

Reply:
This was a copy\paste on my part and the catch has been removed.

I removed it from here and the original location as well because it shouldn't be there at all (especially without a comment explaining why).

---
In reply to: [440857524](https://github.com/microsoft/sarif-visualstudio-extension/pull/143#discussion_r440857524) [](ancestors = 440857524)